apps/backend/src/app/api/latest/internal/external-db-sync/poller/route.ts

```diff
@@ -0,0 +1,129 @@
+import { upstash } from "@/lib/upstash";
+import { globalPrismaClient, retryTransaction } from "@/prisma-client";
+import { createSmartRouteHandler } from "@/route-handlers/smart-route-handler";
+import { OutgoingRequest } from "@prisma/client";
+import {
+  yupBoolean,
+  yupNumber,
+  yupObject,
+  yupString,
+  yupTuple,
+} from "@stackframe/stack-shared/dist/schema-fields";
+import { getEnvVariable, getNodeEnvironment } from "@stackframe/stack-shared/dist/utils/env";
+import { StatusError } from "@stackframe/stack-shared/dist/utils/errors";
+import { wait } from "@stackframe/stack-shared/dist/utils/promises";
+
+export const GET = createSmartRouteHandler({
+  metadata: {
+    summary: "Poll outgoing requests and push to QStash",
+    description:
+      "Internal endpoint invoked by Vercel Cron to process pending outgoing requests.",
+    tags: ["External DB Sync"],
+    hidden: true,
+  },
+  request: yupObject({
+    auth: yupObject({}).nullable().optional(),
+    method: yupString().oneOf(["GET"]).defined(),
+    headers: yupObject({
+      authorization: yupTuple([yupString()]).defined(),
+    }).defined(),
+    query: yupObject({}).defined(),
+  }),
+  response: yupObject({
+    statusCode: yupNumber().oneOf([200]).defined(),
+    bodyType: yupString().oneOf(["json"]).defined(),
+    body: yupObject({
+      ok: yupBoolean().defined(),
+      requests_processed: yupNumber().defined(),
+    }).defined(),
+  }),
+  handler: async ({ headers }) => {
+    const authHeader = headers.authorization[0];
+    if (authHeader !== `Bearer ${getEnvVariable("CRON_SECRET")}`) {
+      throw new StatusError(401, "Unauthorized");
+    }
+
+    const startTime = performance.now();
+    const maxDurationMs = 2 * 60 * 1000;
```

Comment:
thinking about this, maybe we should set it to 3min, because in the worst case vercel cron might schedule the calls 2min apart of each other (start of minute :05 and end of minute :06)